src/gui/folderman.cpp

```diff
@@ -1507,9 +1518,13 @@ void FolderMan::editFileLocally(const QString &userId, const QString &relPath, c
         return;
     }
 
-    const auto checkTokenForEditLocally = new SimpleApiJob(accountFound->account(), QStringLiteral("/ocs/v2.php/apps/files/api/v1/openlocaleditor/%1").arg(token));
+    // Sanitise the token
+    const auto encodedToken = QString(QUrl::toPercentEncoding(token));
+    // Sanitise the relPath
+    const auto encodedRelPath = QUrl::toPercentEncoding(relPath);
```

Comment:
can you also block any path that is not equal to its canonical version ?
so convert it to a canonical path and checks that the input path and the canonical paths are equal ?
that would probably remove the possibility to do many tricks on the file path itself

Reply:
Done, path is now checked against a "cleaned" path for equality